### PR TITLE
Added DatabaseFeatures.rounds_to_even.

### PR DESCRIPTION
**Pull request by timgraham**
This is [useful with MongoDB](https://www.mongodb.com/docs/manual/reference/operator/aggregation/round/#rounding-to-even-values).

### CODE REVIEW

File: django/db/backends/base/features.py [hunk 1]
```diff
@@ -382,6 +382,9 @@ class BaseDatabaseFeatures:
     # SQL to create a model instance using the database defaults.
     insert_test_table_with_defaults = None
 
+    # Does the Round database function round to even?
```
Comment (felixxm) [inline django/db/backends/base/features.py:385]:
```suggestion
    # Does the Round() database function round to even?
```

**Review by felixxm — APPROVED** [review]

**Review by nessita — COMMENTED** [review]
Thank you @timgraham! This looks like a well-isolated and uncontroversial change. However, I feel uneasy about merging this, as in the past, I've seen similar proposals rejected for not being directly used within Django core.

Do you have any references to previous merges that set a precedent for landing a change like this?

**Comment by timgraham** [comment]
I don't think it's controversial to add feature flags to support third-party database backends, so that they don't have to use a heavily forked version of the test suite. (Consider the possibility of moving a built-in backend like Oracle to a third-party package... I wouldn't propose to remove all of the Oracle feature flags in the test suite.)  Certainly there could be some esoteric behaviors that impose too high a maintenance burden in the test suite, but I would say flags like this one are largely up to the third-party backends to keep up to date since patches with new tests in Django can't consider every third-party backend.

I upstreamed nearly all of the changes I needed for CockroachDB. Here are a few:
a7b4a04d6c54679cb0fbc8679367848bd7dae718
227d0c7365cfd0a64d021cb9bdcf77bed2d3f170
f59a2b730685fc62c5cb44101f54faf8921d9bc7
ede9fac75807fe5810df66280a60e7068cc97e4a

I don't anticipate being able to do the same for MongoDB since many of the changes are not appropriate, but this seemed like an easily described behavior that a SQL database could have.

**Comment by felixxm** [comment]
> Do you have any references to previous merges that set a precedent for landing a change like this?

We always tried to encourage folks to build new database backends as a third party solutions. Assuring them that making small tweaks to the base backend to make their life easier is not a problem.

**Comment by nessita** [comment]
Thank you both, this information is really helpful!

**Comment by pauloxnet** [comment]
> > Do you have any references to previous merges that set a precedent for landing a change like this?
> 
> We always tried to encourage folks to build new database backends as a third party solutions. Assuring them that making small tweaks to the base backend to make their life easier is not a problem.

This is something that we should share publicly (e.g. in the database documentation) to encourage folks that want to build new database backends.

**Pull request merged**